### DISCUSSION TITLE
I can't get this plugin to work

### DISCUSSION THREAD

**Discussion by tepene** [vscode-neovim/vscode-neovim#1822] [Q&A] [answered]
I would love to use this extension but I can't get it to work. I must have misunderstood something.

My setup: **NVIM v0.9.5** /  **VSCode 1.86.2** / **vscode-neovim v1.6.1**

I started with a simple `init.lua` provided in the [README](https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#neovim-configuration). Just to get started.

```lua
if vim.g.vscode then
    print("nvim-vscode")
else
    print("nvim")
end
```
My `settings.json` for VSCode has:

```json
{
  // NeoVim
  "extensions.experimental.affinity": {
    "asvetliakov.vscode-neovim": 1
  },
  "vscode-neovim.logLevel": "info",
  "vscode-neovim.logOutputToConsole": true
}
```
Now, when I open a document in VSCode i see the following in the dedug log:

```log
2024-02-19 21:06:36.753 [info] MainController: starting nvim: nvim -N --embed --cmd source /var/home/stephan/.vscode/extensions/asvetliakov.vscode-neovim-1.6.1/vim/vscode-neovim.vim -S /var/home/stephan/.vscode/extensions/asvetliakov.vscode-neovim-1.6.1/runtime/lua/vscode-neovim/force-options.lua -c cd /var/home/stephan/Code/playground
2024-02-19 21:06:37.161 [info] 
nvim
```

which for a start tells me, that the `vim.g.vscode` statement is not working. Anyhow, since I don't configure anything at the moment I went ahead to test basic functionality. If I go into `INSERT` mode I see:
![image](https://github.com/vscode-neovim/vscode-neovim/assets/54766327/94e313f6-4ba1-4bed-8206-b6c41e391478)
in my VSCode status bar. But I can't insert anything into the document. Same goes for `VISUAL` mode. I see
![image](https://github.com/vscode-neovim/vscode-neovim/assets/54766327/31693876-eacf-4878-83a1-f6bdbaed71f1)
but i can't highlight anything. The cursor is not moving at all.

Can someone help me to get this working? Am I missing something? Are there better ways to debug this?

Thanks. Any help is appreciated.

**Comment by justinmk** [comment]
> which for a start tells me, that the `vim.g.vscode` statement is not working.

Seems like your `init.lua` isn't found or applied. In the next release, https://github.com/vscode-neovim/vscode-neovim/pull/1797 will add more info to the logs which might help with this.



> but i can't highlight anything. The cursor is not moving at all.

Maybe some vscode mappings are interfering. Try removing your vscode shortcuts (rename/backup your `keybindings.json` file) and/or disabling all other vscode plugins.

**Reply by tepene** [reply]
thanks @justinmk, but removing `keybindings.json` and disabling all other plugins did not change anything.

**Reply by justinmk** [reply]
Try the "clean" setting: https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#troubleshooting

That won't help find your `init.lua`, but it hopefully fixes the "cursor not moving" issue.

**Reply by tepene** [reply]
![image](https://github.com/vscode-neovim/vscode-neovim/assets/54766327/f9649f80-b071-4834-98e2-0f173aa64fc5)

Unfortunately not. Can't move the cursor in `clean` mode as well

**Reply by tepene** [reply]
ah ok, you led me on the right path. out of curiosity I downloaded the latest release as bin and modified my settings:

```json
"vscode-neovim.neovimExecutablePaths.linux": "/var/home/stephan/Downloads/nvim-linux64/bin/nvim"
```

Now it looks way better. I can move the cursor and insert mode is looking good as well.

So it must have to do with my neovim installation. I have neovim installed in a [bluefin-cli](https://github.com/ublue-os/bluefin/pkgs/container/bluefin-cli#bluefin-cli) container with `brew install neovim` and exported it to the host with `distrobox-export --bin /home/linuxbrew/.linuxbrew/bin/nvim --export-path ~/.local/bin`. I'll have to investigate that further.

**Reply by tepene** [reply, marked as answer]
Alright, I can't find the exact problem with my setup at the moment. So I decided to go with the [AppImage](https://github.com/neovim/neovim/blob/master/INSTALL.md#appimage-universal-linux-package) NeoVim version.

I modified my `settings.json` in VSCode as follows:

```json
{
  // NeoVim Extension
  "extensions.experimental.affinity": {
    "asvetliakov.vscode-neovim": 1
  },
  "vscode-neovim.neovimExecutablePaths.linux": "/var/home/stephan/Code/__devcontainer/apps/nvim.appimage",
  "vscode-neovim.logOutputToConsole": true,
  "vscode-neovim.logLevel": "info"
}
```

This works for me. So, when in VSCode it uses the AppImage and when i use NeoVim directly I use my with `brew` installed version.

Thanks again @justinmk for giving me some hints!!

**Comment by justinmk** [comment]
> So it must have to do with my neovim installation. I have neovim installed in a [bluefin-cli](https://github.com/ublue-os/bluefin/pkgs/container/bluefin-cli#bluefin-cli) container

Thanks for the followup. If you can find a way that vscode-neovim could detect this situation (i.e. some Nvim state or logs that can be checked), please create an issue.

We have created similar checks, for example: https://github.com/vscode-neovim/vscode-neovim/pull/1816
